RxCocoa/Common/DelegateProxy.swift

```diff
@@ -142,17 +143,18 @@ open class DelegateProxy : _RXDelegateProxy {
     - returns: Observable sequence of arguments passed to `selector` method.
      */
     open func methodInvoked(_ selector: Selector) -> Observable<[Any]> {
+        MainScheduler.ensureExecutingOnScheduler()
```

Comment:
Just out of curiosity, do you think it's mandatory to assert execution on the main, or I missing something :)